lib/sda/tree_test.go

```diff
@@ -241,6 +242,21 @@ func TestBigNaryTree(t *testing.T) {
 	}
 }
 
+func TestTreeIsColored(t *testing.T) {
+	dbg.TestOutput(testing.Verbose(), 4)
+	names := []string{"local1:1000", "local1:1001", "local2:1000", "local2:1001"}
+	peerList := genEntityList(tSuite, names)
+	tree := peerList.GenerateBigNaryTree(3, 13)
+	root := tree.Root
+	rootHost, _, _ := net.SplitHostPort(root.Entity.Addresses[0])
+	for _, child := range root.Children {
+		childHost, _, _ := net.SplitHostPort(child.Entity.Addresses[0])
+		if rootHost == childHost {
```

Comment:
If the address is different, so will be the entity-ID.

If the first level is good, we can suppose that the other levels are good, too. Here we have two hosts `local1` and `local2`, so if the root is on `local1`, all it's children will be on `local2`.